comm/src/main/scala/coop/rchain/p2p/p2p.scala

```diff
@@ -47,85 +47,125 @@ case object NetworkAddress {
 
 object Network extends ProtocolDispatcher[java.net.SocketAddress] {
 
-  type KeysStore[F[_]] = Kvs[F, PeerNode, Array[Byte]]
+  type KeysStore[F[_]]    = Kvs[F, PeerNode, Array[Byte]]
+  type ErrorHandler[F[_]] = ApplicativeError_[F, CommError]
 
   import NetworkProtocol._
   import Encryption._
 
+  val defaultTimeout: Duration = Duration(500, MILLISECONDS)
+
   def unsafeRoundTrip[F[_]: Capture: Communication]
     : (ProtocolMessage, ProtocolNode) => CommErr[ProtocolMessage] =
     (pm: ProtocolMessage, pn: ProtocolNode) => {
-      val result = Communication[F].roundTrip(pm, pn)
+      val result = Communication[F].roundTrip(pm, pn, defaultTimeout)
       Capture[F].unsafeUncapture(result)
     }
 
-  def connect[F[_]: Capture: Monad: Log: Time: Metrics: Communication: Encryption](peer: PeerNode)(
-      implicit
-      keysStore: KeysStore[F],
-      err: ApplicativeError_[F, CommError]): F[Unit] = {
+  def findAndConnect[
+      F[_]: Capture: Monad: Log: Time: Metrics: Communication: Encryption: KeysStore: ErrorHandler]
+    : Int => F[Int] =
+    (lastCount: Int) =>
+      (for {
+        _              <- IOUtil.sleep[F](5000L)
```

Comment:
Why do we sleep here for 5 seconds?

Reply:
It's P2P. You let it collect/discover new nodes and then you connect to it in chunks. But to be honest, I don't know really. It was here before :) when I joined.